app/views/supplementary/summary/additional_documentation_section.scala.html

```diff
@@ -14,34 +14,33 @@
  * limitations under the License.
  *@
 
-@import forms.supplementary.DocumentsProduced
+@import models.declaration.supplementary.DocumentsProducedData
 @import models.viewmodels.HtmlTableRow
-
-@(additionalDocumentation: Option[DocumentsProduced])(implicit messages: Messages)
+@(additionalDocumentationData: Option[DocumentsProducedData])(implicit messages: Messages)
 
 @components.summary_list(Some(messages("supplementary.summary.additionalDocumentation.header"))) {
     @components.table_row_no_change_link(HtmlTableRow(
         label = messages("supplementary.summary.additionalDocumentation.documentTypeCode"),
-        value = additionalDocumentation.flatMap(_.documentTypeCode)
+        value = additionalDocumentationData.map(_.documents.map(_.documentTypeCode))
```

Comment:
The first `map` should be `flatMap`.

It's difficult to say it based on the github code, but... maps through the options must return Option[Option[???]]

Reply:
If I change it to a `FlatMap` I actually have a compilation error. It works in this scenario :)

Reply:
I check the type of this element. It's `Option[Seq[Option[String]` whole this: `additionalDocumentationData.map(_.documents.map(_.documentTypeCode))`

We're handling this because HtmlTableRow can take `value: Option[_]`. So flatMap is wrong, just for your information.